src/states/elder/mod.rs

```diff
@@ -665,40 +665,23 @@ impl Elder {
         }
     }
 
+    // Send NodeApproval to the current candidate which promotes them to Adult and allows them to
+    // passively participate in parsec consensus (that is, they can receive gossip and poll
+    // consensused blocks out of parsec, but they can't vote yet)
     fn handle_candidate_approval(
         &mut self,
         new_pub_id: PublicId,
         new_client_auth: Authority<XorName>,
-        outbox: &mut dyn EventBox,
-    ) -> Result<(), RoutingError> {
-        // Once the joining node joined, it may receive the vote regarding itself.
-        // Or a node may receive CandidateApproval before connection established.
-        // If we are not connected to the candidate, we do not want to add them
-        // to our RT.
-        // This will flag peer as valid if its found in peer_mgr regardless of their
-        // connection status to us.
-        let is_connected = if self.peer_mgr.get_peer(&new_pub_id).is_some() {
```

Comment:
Should we not still make sure we are connected.
If connection did not happen before and we get the parsec event Online consensus, we want to be connected so we can send the gossip?

Reply:
That's a good point!